### PR TITLE
police against issues with writing cyclic data

### PR DESCRIPTION
**Pull request by pjfanning**
relates to https://github.com/FasterXML/jackson-core/pull/1055

### CODE REVIEW

File: csv/src/test/java/com/fasterxml/jackson/dataformat/csv/ser/dos/CyclicDataSerTest.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+    public void testListWithSelfReference() throws Exception {
+        List<Object> list = new ArrayList<>();
+        list.add(list);
+        try {
+            MAPPER.writeValueAsString(list);
+            fail("expected JsonMappingException");
+        } catch (JsonMappingException jmex) {
+            assertTrue("JsonMappingException message is as expected?",
+                    jmex.getMessage().startsWith("Document nesting depth (1001) exceeds the maximum allowed"));
```
[the remark this answers is not in the document]
Comment (pjfanning) [reply csv/src/test/java/com/fasterxml/jackson/dataformat/csv/ser/dos/CyclicDataSerTest.java]: done

File: yaml/src/test/java/com/fasterxml/jackson/dataformat/yaml/ser/dos/CyclicDataSerTest.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+    public void testListWithSelfReference() throws Exception {
+        List<Object> list = new ArrayList<>();
+        list.add(list);
+        try {
+            MAPPER.writeValueAsString(list);
+            fail("expected JsonMappingException");
+        } catch (JsonMappingException jmex) {
+            assertTrue("JsonMappingException message is as expected?",
+                    jmex.getMessage().startsWith("Document nesting depth (1001) exceeds the maximum allowed"));
```
[the remark this answers is not in the document]
Comment (pjfanning) [reply yaml/src/test/java/com/fasterxml/jackson/dataformat/yaml/ser/dos/CyclicDataSerTest.java]: done

**Comment by yawkat** [comment]
toml LGTM

**Comment by pjfanning** [comment]
@cowtowncoder does this look ok to merge?

**Comment by cowtowncoder** [comment]
Yes, sorry, have been swamped with other smaller things -- will look into this now. So I haven't forgotten it, just... been delayed.

**Review by cowtowncoder — APPROVED** [review]

**Pull request merged**